service/javadsl/broker/src/main/java/com/lightbend/lagom/javadsl/broker/TopicProducer.java

```diff
@@ -89,7 +90,7 @@
   public static <Message, Event extends AggregateEvent<Event>>
       Topic<Message> taggedStreamWithOffset(
           AggregateEventShards<Event> shards,
-          BiFunction<AggregateEventTag<Event>, Offset, Source<Pair<Message, Offset>, ?>>
+          BiFunction<AggregateEventTag<Event>, Offset, Source<Pair<Optional<Message>, Offset>, ?>>
               eventStream) {
     return new TaggedOffsetTopicProducer<>(shards.allTags(), eventStream);
   }
```

Comment:
The changes in `TopicProducer` are the main problem in this PR as these three edits break the API. Instead, we could consider _adding_ methods as alternatives to the existing ones. I would also mark the new methods as `@ApiMayChange` and keep the original methods as non-deprecated.

Finally, I'd add a comment on the new methods to this PR or #844 so users finding the API could locate and comment on the discussion.